### DISCUSSION TITLE
FX now uses Go extractor

### DISCUSSION THREAD

**Discussion by tv21** [blackjack4494/youtube-dlc#209] [General]
I am waiting for https://github.com/blackjack4494/youtube-dlc/issues/124 to be resolved before I install youtube-dlc but in the meantime I wanted to refer you to the patches in https://github.com/ytdl-org/youtube-dl/pull/26826/files although they do not fully fix the issue.  The patches are as follows (these are to the original, not necessarily your fork, and they are not mine, I am just referencing them):

In youtube_dl/extractor/extractors.py:

Remove the line `from .fxnetworks import FXNetworksIE` at or near line 397.

Remove the file  youtube_dl/extractor/fxnetworks.py completely.

And in the file youtube_dl/extractor/go.py:

```
@@ -38,13 +38,17 @@ class GoIE(AdobePassIE):
        'disneynow': {
            'brand': '011',
            'resource_id': 'Disney',
-        }
+        },
+        'fxnow.fxnetworks': {
+            'brand': '025',
+            'requestor_id': 'dtci',
+        },
    }
    _VALID_URL = r'''(?x)
                    https?://
                        (?:
                            (?:(?P<sub_domain>%s)\.)?go|
-                            (?P<sub_domain_2>abc|freeform|disneynow)
+                            (?P<sub_domain_2>abc|freeform|disneynow|fxnow\.fxnetworks)
                        )\.com/
                        (?:
                            (?:[^/]+/)*(?P<id>[Vv][Dd][Kk][Aa]\w+)|
@@ -99,6 +103,19 @@ class GoIE(AdobePassIE):
            # m3u8 download
            'skip_download': True,
        },
+    }, {
+        'url': 'https://fxnow.fxnetworks.com/shows/better-things/video/vdka12782841',
+        'info_dict': {
+            'id': 'VDKA12782841',
+            'ext': 'mp4',
+            'title': 'First Look: Better Things - Season 2',
+            'description': 'md5:fa73584a95761c605d9d54904e35b407',
+        },
+        'params': {
+            'geo_bypass_ip_block': '3.244.239.0/24',
+            # m3u8 download
+            'skip_download': True,
+        },
    }, {
        'url': 'http://abc.go.com/shows/the-catch/episode-guide/season-01/10-the-wedding',
        'only_matching': True,
```
I apologize that this is not proper patch file format, I don't know how you get one from github so I had to manually type the + and - symbols at the start of the lines.  Anyway this all came from https://github.com/ytdl-org/youtube-dl/pull/26826 but the problem is that it doesn't work on things that require a login.  As I said there,

> I very much appreciate the effort here, and using your example URL it does work, but I find that with these patches installed I still get an error on other types of URL's: ERROR: Unable to download webpage: HTTP Error 401: Unauthorized (caused by <HTTPError 401: 'Unauthorized'>); please report this issue on https://yt-dl.org/bug ... It seems to affect episodes picked from the page at https://www.fxnetworks.com/ that include season and episode numbers in the URL. For example the first episode on that page tonight was https://fxnow.fxnetworks.com/shows/a-wilderness-of-error/episode-guide/season-01/episode-05-a-wilderness-of-error/vdka20447879 and that doesn't work. Your example in your code follows the format:
> 
> https://fxnow.fxnetworks.com/shows/show-name/video/vdka########
> 
> What doesn't work is
> 
> https://fxnow.fxnetworks.com/shows/show-name/episode-guide/season-0#/episode-0#-episode-title/vdka########
> 
> (the #'s represent digits)

The guy that wrote the patch said the issue was that I was not supplying credentials, but in fact I was, and other Go services that require credentials work fine.  My guess is it just isn't parsing the second form of the URL correctly but I don't know for sure.  If there is some way you can figure out why his patch only works for videos that don't require a provider login it would be much appreciated, but also I have a feeling that even his original patches will not make it into the main distribution any time soon, so it would be helpful if you could get them into yours if there's not some reason not to.

**Comment by raleeper** [comment]
For FX, but not other channels in go.py, there are some extra token parameters that are calculated in JavaScript and passed along during the login. I haven't been able to figure out how to create them. But I did find that I could login with a browser and save a cookie file for FX that will work without going through the adobepass logic. Unfortunately, go.py always tries to call adobepass and fails with the 401 error.  You can fix that by changing 

`if video_data.get('accesslevel') == '1':`

to

`if video_data.get('accesslevel') == '1' and self._downloader.params.get('cookiefile') is None:`

in go.py. With that change adobepass login is skipped when a cookie file is provided.
